### PR TITLE
Fix minor issues found by fuzz testing

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@wenningerk , can you review this one? Thanks

File: lib/common/iso8601.c [hunk 1]
```diff
@@ -954,16 +970,17 @@ parse_date(const char *date_str)
...
-    rc = sscanf(date_str, "%d-W%d-%d", &year, &week, &day);
+    rc = sscanf(date_str, "%" SCNu32 "-W%" SCNu32 "-%" SCNu32,
+                &year, &week, &day);
     if (rc == 3) {
-        if (week > crm_time_weeks_in_year(year)) {
+        if ((week < 1U) || (week > crm_time_weeks_in_year(year))) {
             crm_err("'%s' is not a valid ISO 8601 date/time specification "
-                    "because '%d' is not a valid week of the year (max %d)",
-                    date_str, week, crm_time_weeks_in_year(year));
+                    "because '%" PRIu32 "' is not a valid week of the year "
```
Comment (wenningerk) [inline lib/common/iso8601.c:993]: we could give the year here as well

File: lib/common/strings.c [hunk 2]
```diff
@@ -457,7 +457,8 @@ pcmk_parse_interval_spec(const char *input, guint *result_ms)
         crm_time_t *period_s = crm_time_parse_duration(input);
 
         if (period_s != NULL) {
-            msec = 1000 * crm_time_get_seconds(period_s);
+            msec = crm_time_get_seconds(period_s);
+            msec = QB_MIN(msec, G_MAXUINT / 1000) * 1000;
```
Comment (wenningerk) [inline lib/common/strings.c:461]: would we exceed 80 chars when writing crm_time_get_seconds(period_s) directly in QB_MIN?
at least we would never write seconds in a variable that is called msec ;-)
and the casting is not the reason as it is already long long.
picky - was just thinking why you did it the way you did ...

**Review by wenningerk — COMMENTED** [review]
Looking good - nothing really apart from I think checking day for 0

File: lib/common/iso8601.c [hunk 3]
```diff
@@ -913,37 +914,52 @@ parse_date(const char *date_str)
...
+                    date_str, month);
+            goto invalid;
+        } else if ((year < 1U) || (year > INT_MAX)) {
             crm_err("'%s' is not a valid ISO 8601 date/time specification "
-                    "because '%d' is not a valid month", date_str, month);
+                    "because '%" PRIu32 "' is not a valid year",
+                    date_str, year);
             goto invalid;
-        } else if (day > crm_time_days_in_month(month, year)) {
+        } else if ((day > INT_MAX)
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/iso8601.c:950]: no, good catch

File: lib/common/iso8601.c (hunk 1 above)
Comment (kgaillot) [reply lib/common/iso8601.c:993]: good point

**Comment by kgaillot** [comment]
@wenningerk , added commit on top to address review

File: lib/common/strings.c (hunk 2 above)
Comment (wenningerk) [reply lib/common/strings.c:461]: I'd assume the compiler optimizes that away - but yeah ...

**Comment by wenningerk** [comment]
Looking good

**Pull request merged**